ToNote/ViewModels/MainViewModel.cs

```diff
@@ -63,8 +63,12 @@ public ICommand AddNoteCommand
 
                     var result = DialogService.OpenDialog(dialog);
 
-                    if (result != null && !Notes.Any(x => x.Name.Equals(result.Name)))
+                    var forbiddenChars = "\\/:?<>|".ToCharArray();
+
+                    if (result.Name.IndexOfAny(forbiddenChars) == -1 && result != null && !Notes.Any(x => x.Name.Equals(result.Name)))
```

Comment:
You don't check if the result was null, which means right now it crashes if you close the Add Note dialog with the exit button.